x/amm/keeper/farming.go

```diff
@@ -138,7 +138,13 @@ func (k Keeper) TerminateFarmingPlan(ctx sdk.Context, plan types.FarmingPlan) er
 	plan.IsTerminated = true
 	k.SetFarmingPlan(ctx, plan)
 	if plan.IsPrivate {
-		k.SetNumPrivateFarmingPlans(ctx, k.GetNumPrivateFarmingPlans(ctx)-1)
+		numPlans := k.GetNumPrivateFarmingPlans(ctx)
+		if numPlans > 1 {
+			k.SetNumPrivateFarmingPlans(ctx, k.GetNumPrivateFarmingPlans(ctx)-1)
+		} else {
+			k.DeleteNumPrivateFarmingPlans(ctx)
+		}
+	} else {
```

Comment:
Unless this part of the implementation is missing, I think it can be erased

Reply:
Thank you for finding it out!